cleanlab_studio/internal/api/api.py

```diff
@@ -926,20 +932,34 @@ async def wrapper(*args: Any, **kwargs: Any) -> Any:
                 )
                 raise
             except aiohttp.client_exceptions.ClientConnectorError as e:
-                # note: we don't increment num_try here, because we don't want connection errors to count against the total number of retries
-                sleep_time = 2**num_try
+                if num_connection_error_retry == (max_connection_error_retries // 2):
+                    warnings.warn(
+                        f"Connection error after {max_connection_error_retries // 2} retries. Retrying..."
```

Comment:
nit: use `num_connection_error_retry` directly instead of `max_connection_error_retries // 2`?